src/main/java/ognl/OgnlOps.java

```diff
@@ -128,9 +128,11 @@ public static boolean isEqual(Object object1, Object object2)
 
         if (object1 == object2) {
             result = true;
+        } else if (object1 == null || object2 == null) {
+            result = false;
```

Comment:
I think line 131 would still need to remain:
`} else if (object1 == null || object2 == null) {`
otherwise wouldn't the rest of the comparison logic below get skipped ?  
Line 132 could still be removed if desired.

Reply:
Sorry, I wasn't clear enough.
What I meant was ...

```java
        if (object1 == object2) {
            result = true;
        } else if (object1 != null && object2 != null) {
            if (object1.getClass().isArray()) {
                if (object2.getClass().isArray() && (object2.getClass() == object1.getClass())) {
                    result = (Array.getLength(object1) == Array.getLength(object2));
```

Reply:
Hmm, I wanted to avoid adding additional nesting and make it more explicit, but I'll make the change.

Reply:
Maybe I still wasn't clear enough...

```diff
--- a/src/main/java/ognl/OgnlOps.java
+++ b/src/main/java/ognl/OgnlOps.java
@@ -128,9 +128,7 @@ public abstract class OgnlOps implements NumericTypes
 
         if (object1 == object2) {
             result = true;
-        } else if (object1 == null || object2 == null) {
-            result = false;
-        } else {
+        } else if (object1 != null && object2 != null) {
             if (object1.getClass().isArray()) {
                 if (object2.getClass().isArray() && (object2.getClass() == object1.getClass())) {
                     result = (Array.getLength(object1) == Array.getLength(object2));
```

There is no additional nesting. :)

Reply:
Yeah, I understood. I just confused myself. :) I've pushed the change.

Reply:
Hi.

The latest change (as suggested by @harawata) seems to be similar to the logic in OGNL 2.6.9.  I tried to locate any changelog information for why it changed in OGNL 2.7, but that information does not seem to be around (only an announcement mentioning the release of Tapestry 4.1.2 / OGNL 2.7 in June 2007).